plugins/fix-china-map-offset.js

```diff
@@ -0,0 +1,315 @@
+// ==UserScript==
+// @id             iitc-plugin-fix-china-map-offset
+// @name           IITC plugin: Fix maps offsets in China
+// @category       Tweaks
+// @version        0.2.0.@@DATETIMEVERSION@@
+// @description    [@@BUILDNAME@@-@@BUILDDATE@@] Show correct maps for China user by applying offset tweaks.
+@@METAINFO@@
+// ==/UserScript==
+
+@@PLUGINSTART@@
+
+// PLUGIN START ////////////////////////////////////////////////////////
+
+// use own namespace for plugin
+window.plugin.fixChinaOffset = {};
+
+// Before understanding how this plugin works, you should know 3 points:
+//
+//   Point1.
+//     The coordinate system of Ingress is WGS-84.
+//     However, the tiles of Google maps (except satellite map) and some other maps
+//     in China have offsets (base on GCJ-02 coordinate system) by China policy.
+//     That means, if you request map tiles by giving GCJ-02 position, you
+//     will get the correct map.
+//
+//   Point2.
+//     Currently there are no easy algorithm to transform from GCJ-02 to WGS-84,
+//     but we can easily transform data from WGS-84 to GCJ-02.
+//
+//   Point3.
+//     When using, for example, Google maps in IITC, the layer structure looks like this:
+//      --------------------------
+//     |    Other Leaflet layers  |  (Including portals, links, fields, and so on)
+//      --------------------------
+//     | L.GridLayer.GoogleMutant |  (Only for controling)
+//      --------------------------
+//     |      Google Map layer    |  (Generated by Google Map APIs, for rendering maps)
+//      --------------------------
+//
+//     When users are interacting with L.GridLayer.GoogleMutant (for example, dragging, zooming),
+//     L.GridLayer.GoogleMutant will perform the same action on the Google Map layer using Google
+//     Map APIs.
+//
+// So, here is the internal of the plugin:
+//
+// The plugin overwrites behaviours of L.GridLayer.GoogleMutant. When users are dragging the map,
+// L.GridLayer.GoogleMutant will pass offseted positions to Google Map APIs (WGS-84 to GCJ-02).
+// So Google Map APIs will render a correct map.
+//
+// The offset between Google maps and Ingress objects can also be fixed by applying
+// WGS-84 to GCJ-02 transformation on Ingress objects. However we cannot simply know
+// the requesting bounds of Ingress objects because we cannot transform GCJ-02 to
+// WGS-84. As a result, the Ingress objects on maps would be incomplete.
+//
+// The algorithm of transforming WGS-84 to GCJ-02 comes from:
+// https://on4wp7.codeplex.com/SourceControl/changeset/view/21483#353936
+// There is no official algorithm because it is classified information.
+//
+// Here we use the PRCoords implementation of this algorithm, which contains
+// a more careful yet still rough "China area" check in "insane_is_in_china.js".
+// Comments and code style have been adapted, mainly to remove profanity.
+// https://github.com/Artoria2e5/PRCoords
```

Comment:
I suggest to add the link as there is more info about source data of polygon.
```suggestion
// https://github.com/Artoria2e5/PRCoords (more info: https://github.com/iitc-project/ingress-intel-total-conversion/pull/1188)
```